### ISSUE TITLE
Core styles for editor using 4.x class names

### ISSUE THREAD

**Issue by joyously** [ClassicPress-research/ClassicPress-Editor#55]
The wp-includes/css/editor.css file styles the container elements for parts of the editor, the icons on the menu, the character map, and the hover states of several things.
Some of the class names are specific to Tiny 4.x. A lot of class names changed from a `mce` prefix to a `tox` prefix.
There are a few classes mentioned in wp-admin/css/edit.css also. These need to be checked to see if any changed for 5.x.
There is a file just for IE, wp-admin/css/ie.css, which styles the icons.
The admin color schemes (in wp-admin/css/colors) style the background color for the editor menu items.
The wp-admin/css/customize-widgets.css file sets the z-index for parts of the editor for the Text Widget, using old names. 

I wanted to float the idea here that changing this CSS is an opportunity to give the editor a dark mode and/or use CSS custom properties for colors so they aren't hard-coded in this CSS.

**Comment by johnalarcon** [comment]
This is good research, Joy... and you're right, it's going to take some work to get it all in order. Regarding dark mode, this was already built into TinyMCE (it responds to the user's OS light/dark setting) with my original version's config properties:

`skin: useDarkMode ? 'oxide-dark' : 'oxide',`

Perhaps we can still leverage that in the PHP config options.

**Comment by joyously** [comment]
No, doing dark mode at init time doesn't help with themes needing to do it dynamically.

**Comment by johnalarcon** [comment]
Ok. The built-in behavior only styles the editor itself without regard for its surroundings, so, I see what you're saying.

**Comment by joyously** [comment]
The wp-includes/css/editor.css file has styles for `.mce-window` which I determined is used on the Help modal, which is created by `dialog` function, so I'm assuming that's the class for 4.x dialogs. Apparently, there is at least one reference in the lightgray/skin.min.css also. 
The dialog has a child, with mce-window-head element, mce-window-body element, and mce-foot element. A sibling to mce-window is `#mce-modal-block` which is the 70% black backdrop.

The 5.x `dialog`s have changed a little from 4.x. There are more nested elements and the main class is tox-dialog-wrap with a tox-dialog-wrap__backdrop child (which is white) and a tox-dialog child (which then has a header, content, footer).

I'm wondering how much of these styles can be abandoned. There are z-index rules to consider, but all the little details of colors and padding might be unnecessary.

**Comment by joyously** [comment]
The 4.x menu is a popover, is a `ui.FloatPanel`.
```
<div id="mceu_115" class="mce-container mce-panel mce-floatpanel mce-menu mce-menu-sub 
mce-menu-align mce-menu-sub-tr-tl" hidefocus="1" tabindex="-1" role="application" 
style="border-width: 1px; z-index: 100101; left: 371.417px; top: 420.067px; width: 180px; display: none;" 
aria-labelledby="mceu_115" aria-describedby="mceu_115-none">
```


The 5.x menu is ???
It uses much lower z-index.
```
<div class="tox tox-silver-sink tox-tinymce-aux" 
style="position: relative;"><div id="aria-owns_7953174261751631678198027" 
role="listbox">
  <div class="tox-tiered-menu">
    <div role="menu" class="tox-menu tox-collection tox-collection--list tox-selected-menu" 
    style="position: absolute; left: 191px; top: -539.233px; max-height: 176px; overflow: hidden auto; min-width: 107px;" 
    data-alloy-placement="southeast">
      <div class="tox-collection__group">
        <div title="Text" role="menuitem" aria-haspopup="true" aria-expanded="false" tabindex="-1" 
        class="tox-menu-nav__js tox-collection__item" aria-disabled="false">
```
but that doesn't match the CSS I found:
```
.tox .tox-dialog__popups {
	position: absolute;
	width: 100%;
	z-index: 1100;
}
.tox .tox-menu {
	background-color: #fff;
	display: inline-block;
	overflow: hidden;
	vertical-align: top;
	z-index: 1150;
}
```

**Comment by joyously** [comment]
The script wp-includes/js/tinymce/tiny_mce_popup.js has a comment
```
 * TinyMCE popup/dialog helper class. This gives you easy access to the
 * parent editor instance and a bunch of other things. It's higly recommended
 * that you load this script into your dialogs.
```
I already commented out a few things that were breaking the script, but one of the things it does is load `plugins/compat3x/css/dialog.css` so the popups have CSS besides what Tiny does. But when I looked at the CSS to change to the new dialog classes, I didn't recognize anything from 4.x. It almost looks like the old ones were separate windows and the new ones are not, yet there is a 5.x windowManager still. This needs investigation to find what windows are used in 3.x or 4.x dialogs so that can be compared to 5.x.

**Comment by joyously** [comment]
The editor.css file still has some old styles that affect the context toolbar and link dialog. The wplink plugin needs to be working to see how the styles should be changed though.
See #49.
